src/Core/Requests/RequestDataConverter.cs

```diff
@@ -289,9 +290,24 @@ internal object ConvertSingleEntity(ExpandoObject dynamicObject, object dto)
                 _logger.LogDebug("Have target prop? '{0}' with name: '{1}' and actual value: '{2}'", targetProp != null,
                     dynamicProp.Key, dynamicValue);
 
-                // skip any nullable properties with a empty value
-                if (targetProp != null && Nullable.GetUnderlyingType(targetProp.PropertyType) != null && string.IsNullOrEmpty(dynamicValue.ToString()))
-                    continue;
+                if (targetProp != null)
+                {
+                    // property type
+                    var pt = targetProp.PropertyType;
+                    var val = dynamicValue.ToString();
+
+                    // skip any nullable properties with a empty value
+                    if (Nullable.GetUnderlyingType(pt) != null && string.IsNullOrEmpty(val))
+                        continue;
+
+                    // convert to DateTime/DateTime?
+                    if ((pt == typeof(DateTime) || pt == typeof(DateTime?)) && long.TryParse(val, out var milliseconds))
```

Comment:
**question** are we sure that the timestamp coming back from HubSpot is in `ms` and not in `s` ?

Reply:
Yes, i checked doc: [HubSpot API endpoints accept UNIX formatted timestamps in milliseconds.](https://legacydocs.hubspot.com/docs/faq/how-should-timestamps-be-formatted-for-hubspots-apis)